pcap/pcap.go

```diff
@@ -336,8 +379,27 @@ func (p *Handle) getNextBufPtrLocked(ci *gopacket.CaptureInfo) error {
 	}
 	var result NextError
 	for {
+		// test if we need to stop
+		select {
+		case <-p.stop:
+			return io.EOF
+		default:
+		}
+
+		if p.timeout < 0 {
```

Comment:
to be a little more self-documenting, let's say:  `if p.timeout == BlockForever`